docs/contributing/code-reviewing.md

```diff
@@ -1,231 +1,364 @@
-reviewers also need focused chunks of time to write code and do other
-things, so an immediate reply isn't always possible. But a good
-benchmark is to try to always reply **within one workday**, at least
-with a short initial reply, if you're working regularly on Zulip. And
-sooner is better.
-
-## Things to look for
-
-- _The CI build._ The tests need to pass. One can investigate
-  any failures and figure out what to fix by clicking on a red X next
-  to the commit hash or the Detail links on a pull request. (Example:
-  in [#17584](https://github.com/zulip/zulip/pull/17584),
-  click the red X before `49b10a3` to see the build jobs
-  for that commit. You can see that there are 7 build jobs in total.
-  All the 7 jobs run in GitHub Actions. You can see what caused
-  the job to fail by clicking on the failed job. This will open
-  up a page in the CI that has more details on why the job failed.
-  For example [this](https://github.com/zulip/zulip/runs/2092955762)
-  is the page of the "Debian 10 Buster (Python 3.7, backend + frontend)" job.
-  See our docs on [continuous integration](../testing/continuous-integration.md)
-  to learn more.
-
-- _Technical design._ There are a lot of considerations here:
-  security, migration paths/backwards compatibility, cost of new
-  dependencies, interactions with features, speed of performance, API
-  changes. Security is especially important and worth thinking about
-  carefully with any changes to security-sensitive code like views.
-
-- _User interface and visual design._ If frontend changes are
-  involved, the reviewer will check out the code, play with the new
-  UI, and verify it for both quality and consistency with the rest of
-  the Zulip UI. We highly encourage posting screenshots to save
-  reviewers time in getting a feel for what the feature looks like --
-  you'll get a quicker response that way.
+## How to review code
+
+Whether you are reviewing your own code or somebody else's, this section
+describes how to go about it.
+
+If you are reviewing somebody else's code, you will likely need to first fetch
+it so that you can play around with the new functionality. If you're working in
+the Zulip server codebase, use our [git tool] `tools/fetch-rebase-pull-request`
+to check out a pull request locally and rebase it onto `main`.
+
+### Code review checklist
+
+The following review steps apply to the majority of PRs.
+
+**Think about the issue:**
+
+1. Start by **rereading the issue** the PR is intended to solve. Are you
+   confident that you **understand everything the issue is asking for**? If not,
+   try exploring the relevant parts of the Zulip app and reading any linked
+   discussions on the [development community server][czo] to see if the
+   additional context helps. If any part is still confusing, post a GitHub
+   comment or a message on the [development community server][czo] explaining
+   precisely what points you find confusing.
```

Comment:
I would consider adding a sentence that "If it is difficult to determine whether the pull request correctly implements the issue, that's likely a sign the pull request is missing important explanation of the changes."